sshd-common/src/main/java/org/apache/sshd/common/cipher/BuiltinCiphers.java

```diff
@@ -102,6 +104,31 @@ public Cipher create() {
      */
     @Deprecated
     blowfishcbc(Constants.BLOWFISH_CBC, 8, 0, 16, "Blowfish", 128, "Blowfish/CBC/NoPadding", 8),
+    cc20p1305_openssh(Constants.CC20P1305_OPENSSH, 8, 16, 64, "ChaCha", 256, "ChaCha", 8) {
+        private volatile Boolean supported;
+
+        @Override
+        public boolean isSupported() {
+            if (supported == null) {
+                synchronized (this) {
+                    if (supported == null) {
+                        try {
+                            SecurityUtils.getCipher("ChaCha");
+                            supported = true;
+                        } catch (GeneralSecurityException ignored) {
+                            supported = false;
+                        }
+                    }
+                }
+            }
+            return supported;
+        }
```

Comment:
If this works only with BC then you can replace/add `SecurityUtils.isBouncyCastleRegistered()` call here

Reply:
I just came across that method today as well. This could technically be supported by just copying the source code for ChaCha20 and Poly1305 since I had to combine them into the OpenSSH-specific AEAD form.